### PR TITLE
[MLP-76] Add mongodb config, driver and connection

### PR DESCRIPTION
**Pull request by matiasbeckerle**
# Background

Yep. This one is short, enjoy!

There is the intention to accomplish the **m** part of MERN so basically this aims to only add the basic infrastructure to support mongodb.

Next PR will contain all the User-related changes.

# Changes

- Added a mongodb server file which connects to the server and provides a mongodbConnection to use in other files/classes.
- Removed old code from `server.js` related to mongodb.
- Added `config` package so we can take advantage of configurations. Documented a little bit and added some example values.

### CODE REVIEW

**Comment by matiasbeckerle** [comment]
As usual, if you have time, your feedback is highly appreciated @AlphaGit.

**Review by mravinale — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -91,14 +91,15 @@
     "body-parser": "^1.16.1",
     "bootstrap": "^3.3.7",
     "boron": "^0.2.3",
+    "config": "^1.25.1",
```
Comment (AlphaGit) [inline package.json]: Is this package used at all? If it is, shouldn't it be a dev dependency?

Comment (matiasbeckerle) [inline package.json]: Yes, it is used here https://github.com/MakingSense/mern-seed/pull/17/files#diff-ce90ba09abeb73becc09bc91a8b8c04fR2.

No, it shouldn't be a *dev* dependency, it is used in the server side code (api).

**Review by AlphaGit — APPROVED** [review]
Thanks, looks great!

File: package.json (hunk 1 above)
Comment (AlphaGit) [inline package.json]: Agreed -- I missed that. For some reason I saw the `import x from y` format and I completely forgot you could import packages that way too. (I thought you were importing the json file directly.)

Comment (matiasbeckerle) [inline package.json]: Np dude!

**Review by juampick — APPROVED** [review]
Excellent work @matiasbeckerle. Thanks

**Comment by matiasbeckerle** [comment]
Appreciated folks!

**Pull request merged**